### PR TITLE
[Layer settings tlbx] remove the empty space when no data

### PR DESCRIPTION
**Pull request by mathildemerle**
:zzz: 

As title says : remove the empty space when there is no data in the container.

:sleeping:

### CODE REVIEW

File: src/medCore/gui/medAbstractWorkspace.cpp [hunk 1]
```diff
@@ -105,11 +105,11 @@ medAbstractWorkspace::medAbstractWorkspace(QWidget *parent) : QObject(parent), d
     d->layersToolBox->setTitle("Layer settings");
     d->layersToolBox->hide();
 
-    d->layerListToolBox = new medToolBox(d->layersToolBox);
```
Comment (mikejbuckingham) [inline src/medCore/gui/medAbstractWorkspace.cpp]: Why change this? This introduces a leak.

File: src/medCore/gui/medAbstractWorkspace.cpp [hunk 2]
```diff
@@ -105,11 +105,11 @@ medAbstractWorkspace::medAbstractWorkspace(QWidget *parent) : QObject(parent), d
...
+    d->layerListToolBox = new medToolBox();
     d->layerListToolBox->header()->hide();
     d->layersToolBox->addWidget(d->layerListToolBox);
 
-    d->interactorToolBox = new medToolBox(d->layersToolBox);
```
Comment (mikejbuckingham) [inline src/medCore/gui/medAbstractWorkspace.cpp]: Also here.

File: src/medCore/gui/medAbstractWorkspace.cpp [hunk 3]
```diff
@@ -105,11 +105,11 @@ medAbstractWorkspace::medAbstractWorkspace(QWidget *parent) : QObject(parent), d
...
+    d->interactorToolBox = new medToolBox();
```
Comment (mikejbuckingham) [inline src/medCore/gui/medAbstractWorkspace.cpp]: And here.

Remember that d is just a POD struct, there's no cleanup. We might delete the pod, but we don't free any memory tracked by the pointers in the POD.

**Comment by mathildemerle** [comment]
Parent has been added :)

**Pull request merged**